Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -630,36 +631,58 @@ open class RouteController: NSObject {
                       category: .navigation)
         }
 
+        self.navigatorType = Navigator.self
+        self.indexedRouteResponse = indexedRouteResponse
+        self.dataSource = source
+
         var isRouteOptions = false
         if case .route = indexedRouteResponse.routeResponse.options {
             isRouteOptions = true
         }
         let options = indexedRouteResponse.validatedRouteOptions
-        Navigator.datasetProfileIdentifier = options.profileIdentifier
-        
+
+        self.routeProgress = RouteProgress(route: indexedRouteResponse.currentRoute!, options: options)
+        self.refreshesRoute = isRouteOptions && options.profileIdentifier == .automobileAvoidingTraffic && options.refreshingEnabled
+
+        super.init()
+
+        commonInit(customRoutingProvider: customRoutingProvider, options: options)
+
+        Self.instanceLock.lock()
+        Self.instance = self
+        Self.instanceLock.unlock()
+    }
+
+    init(indexedRouteResponse: IndexedRouteResponse,
```

Comment:
This init should also have a check for duplicate instances, for consistency reasons